src/screens/Profile/ForumNotificationsScreen.js

```diff
@@ -0,0 +1,198 @@
+import AsyncLock from 'async-lock';
+import Notification from './Notification';
+import DefaultNotifications from './DefaultNotifications';
+
+export default class ForumNotificationsScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { notifications: [], loading: true, visible: false };
+    this.setNotificationsToViewed = this.setNotificationsToViewed.bind(this);
+    this.setVisibleFalse = this.setVisibleFalse.bind(this);
+    this.setVisibleTrue = this.setVisibleTrue.bind(this);
+    this.onClearAsyncNotifications = this.onClearAsyncNotifications.bind(this);
+    this.lock = new AsyncLock();
+  }
+
+  componentDidMount() {
+    /* TODO: Make each notification connected to the userID for who it should be sent to */
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+
+    this.unsubscribeFromFirestore = firestore().collection('profile_notifications')
+      .orderBy('createdAt', 'desc')
+      .onSnapshot((snapshot) => {
+        const newNotificationsSnapshot = snapshot.docChanges().filter((change) => change.type === 'added');
+        const newNotificationDocs = newNotificationsSnapshot.map((change) => change.doc);
+        const allNotifications = newNotificationDocs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        const { currentUserID } = this.state;
+
+
+        for (let index = 0; index < allNotifications.length; index += 1) {
+          const date = allNotifications[index].createdAt
+            ? allNotifications[index].createdAt.toDate() : null;
+          if (date != null) {
+            allNotifications[index].createdAt = date.toString();
+          }
+        }
+        if (allNotifications.length > 0) {
+          this.lock.acquire(currentUserID, async () => {
+            await AsyncStorage.getItem(currentUserID).then(async (value) => {
+              if (value) {
+                const storageValue = allNotifications.concat(JSON.parse(value));
+                await AsyncStorage.setItem(currentUserID, JSON.stringify(storageValue));
+                const newNotificationDocRefs = newNotificationsSnapshot.map(
+                  (change) => change.doc.ref,
+                );
+                newNotificationDocRefs.forEach((docRef) => docRef.delete());
+              } else {
+                const storageValue = allNotifications;
+                await AsyncStorage.setItem(currentUserID, JSON.stringify(storageValue));
+                const newNotificationDocRefs = newNotificationsSnapshot.map(
+                  (change) => change.doc.ref,
+                );
+                newNotificationDocRefs.forEach((docRef) => docRef.delete());
+              }
+            });
+          }, (err) => {
+            this.setState({ errorMessage: err, loading: false });
+          });
+        }
+
+        AsyncStorage.getItem(currentUserID).then(async (value) => {
```

Comment:
For me: Necessary because the previous lines update notifications in Async when firebase is not empty. If firebase is empty, you need to get from AsyncStorage, so this line permits both situations: when firebase is and is not empty

Reply:
Is this because of line 50, where we check that `allNotifications.length > 0`? I don't think the condition is necessary. If we remove it, I think we can get away with not making this `getItem` call.